src/pyspiffe/workloadapi/default_workload_api_client.py

```diff
@@ -34,6 +37,51 @@
     WORKLOAD_API_HEADER_VALUE,
 )
 
+# GRPC Error Codes that the client will not retry on:
+#  - INVALID_ARGUMENT is not retried according to the SPIFFE spec because the request is invalid
+#  - CANCELLED is not retried because it occurs when the caller has canceled the operation.
+_NON_RETRYABLE_CODES = {grpc.StatusCode.CANCELLED, grpc.StatusCode.INVALID_ARGUMENT}
+
+
+class _RetryHandler:
```

Comment:
I missed tests for this class.

Reply:
Added tests for that class and all the missing docs.